### PR TITLE
Comment

### PR DESCRIPTION
**Pull request by bhargaviks**
My comments page looks like this now: https://screenshot.googleplex.com/79u59uz6UNo.png
Learnt about datastore and entities. Used them to permanently store my comments, and load it whenever required.

Files that help in doing them and their purpose:
1. comments.html : has the basic outline of the discussion and calls getMesssages() onload to load the comments. it also has a form that when submitted creates a post request to NewCommentServlet.
2.  NewCommentServlet.java : implements the storage of the newly added comment into datastore.
3. DataServlet.java : When the page is loaded, getMessages() is called, and this fetches and triggers this servlet to load all the comments from the datastore.
4. DeleteCommentServlet.java : This servlet takes the id of the comment to be deleted, and deletes it from datastore. 
4. script.js : has getMessages() that loads the comments, and delete comment() to delete a comment. 
5. Comment.java: This is a comment class that I created. Each comment has this as its datatype.

Other changes I made to the page: 
1. Made the page look a little more attractive by involving bulma classes. 
2. Added my bulma footer that is available in all my html pages.

Update: complete week-3 -server step-5-datastore walkthrough with thorough understanding! 
Thank you!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    commentEntity.setProperty("comment", text);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    response.sendRedirect("/comments.html");
+
+    
+  }
+
```
Comment (isaacwdavis) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java:32]: remove extra empty line

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  private final long id;
+  private final String text;
+  private final long timestamp;
+
+  public Comment(long id, String text, long timestamp) {
+    this.id = id;
+    this.text = text;
+    this.timestamp = timestamp;
+  }
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:15]: nit: no newline at EOF

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
+
+    Key commentEntityKey = KeyFactory.createKey("Comment", id);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(commentEntityKey);
+  }
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:26]: Ditto

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java:34]: Ditto

File: portfolio/src/main/webapp/style.css [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  border: thin solid black;
+  display: flex;
+  margin: 10px;
+  padding: 10px;
+  width: 300px;
+}
+
+.comment span {
+  flex-grow: 1;
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/style.css:15]: Ditto

**Pull request merged**